### PR TITLE
Create `polaris/declaration-property-value-disallowed-list` Stylelint rule

### PR DESCRIPTION
**Pull request by aaronccasanova**
This PR introduces a wrapper around the built-in Stylelint `declartion-property-value-disallowed-list` rule to ignore failures in `@font-face` at-rules. The issue is described further in #7991

### CODE REVIEW

File: stylelint-polaris/plugins/typography-declaration-property-value-disallowed-list/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        },
+        (warning) => {
+          // Ignore invalid options as they are already surfaced by Stylelint
+          if (warning.stylelintType === 'invalidOption') return;
+
+          if (
+            warning.node.type === 'decl' &&
+            warning.node.prop === 'font-weight' &&
+            warning.node.parent.type === 'atrule' &&
+            warning.node.parent.name === 'font-face'
```
Comment (aaronccasanova) [inline stylelint-polaris/plugins/typography-declaration-property-value-disallowed-list/index.js]: Open question: Should this rule ignore all failures in `@font-face` rules?

There are [many descriptors](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face#descriptors) in `@font-face` rules and none of which can accept Polaris custom properties.

Comment (sam-b-rose) [reply stylelint-polaris/plugins/typography-declaration-property-value-disallowed-list/index.js]: I would say "yes". Perhaps just remove the `warning.node.prop === 'font-weight' &&` line so we can return early for all `@font-face` properties.

**Review by sam-b-rose — APPROVED** [review]
Looks great-simple and clean! Thanks for adding the test as well. Nice work 💯

**Comment by sam-b-rose** [comment]
Fixes #7991

File: stylelint-polaris/plugins/typography-declaration-property-value-disallowed-list/index.js (hunk 1 above)
Comment (aaronccasanova) [reply stylelint-polaris/plugins/typography-declaration-property-value-disallowed-list/index.js]: Done, thanks! https://github.com/Shopify/polaris/pull/7993/commits/365be4c0326540e75cf2176a02d8d28f8aa0d805

**Review by sophschneider — APPROVED** [review]
Nice bug catch and nice solution! thanks for the test cases, helped to understand the problem :)

**Pull request merged**